test/extensions/filters/http/jwt_authn/group_verifier_test.cc

```diff
@@ -0,0 +1,537 @@
+      .WillRepeatedly(Invoke([](Http::HeaderMap&, std::vector<JwtLocationConstPtr>*,
+                                AuthenticatorCallback callback) { callback(Status::Ok); }));
+  mock_auths_.push_back(std::move(mock_auth));
+  createVerifier();
+  EXPECT_CALL(mock_cb_, onComplete(_)).WillOnce(Invoke([](const Status& status) {
+    ASSERT_EQ(status, Status::Ok);
+  }));
+  auto headers = Http::TestHeaderMapImpl{
+      {"example-auth-userinfo", ""},
+      {"other-auth-userinfo", ""},
+  };
+  context_ = Verifier::createContext(headers, &mock_cb_);
+  verifier_->verify(context_);
+  EXPECT_FALSE(headers.has("example-auth-userinfo"));
+  EXPECT_TRUE(headers.has("other-auth-userinfo"));
+}
+
+TEST_F(GroupVerifierTest, TestRequiresAnyLastAuthOk) {
+  MessageUtil::loadFromYaml(RequiresAnyConfig, proto_config_);
+  auto mock_auth = std::make_unique<MockAuthenticator>();
+  EXPECT_CALL(*mock_auth.get(), doVerify(_, _, _))
+      .WillRepeatedly(
+          Invoke([](Http::HeaderMap&, std::vector<JwtLocationConstPtr>*,
+                    AuthenticatorCallback callback) { callback(Status::JwtUnknownIssuer); }));
+  mock_auths_.push_back(std::move(mock_auth));
+  mock_auth = std::make_unique<MockAuthenticator>();
+  EXPECT_CALL(*mock_auth.get(), doVerify(_, _, _))
+      .WillRepeatedly(Invoke([](Http::HeaderMap&, std::vector<JwtLocationConstPtr>*,
+                                AuthenticatorCallback callback) { callback(Status::Ok); }));
+  mock_auths_.push_back(std::move(mock_auth));
+  createVerifier();
+  EXPECT_CALL(mock_cb_, onComplete(_)).WillOnce(Invoke([](const Status& status) {
+    ASSERT_EQ(status, Status::Ok);
+  }));
+  auto headers = Http::TestHeaderMapImpl{
+      {"example-auth-userinfo", ""},
+      {"other-auth-userinfo", ""},
+  };
+  context_ = Verifier::createContext(headers, &mock_cb_);
+  verifier_->verify(context_);
+  EXPECT_FALSE(headers.has("example-auth-userinfo"));
+  EXPECT_FALSE(headers.has("other-auth-userinfo"));
+}
+
+TEST_F(GroupVerifierTest, TestRequiresAnyAllAuthFailed) {
+  MessageUtil::loadFromYaml(RequiresAnyConfig, proto_config_);
+  auto mock_auth = std::make_unique<MockAuthenticator>();
+  EXPECT_CALL(*mock_auth.get(), doVerify(_, _, _))
+      .WillRepeatedly(
+          Invoke([](Http::HeaderMap&, std::vector<JwtLocationConstPtr>*,
+                    AuthenticatorCallback callback) { callback(Status::JwtHeaderBadKid); }));
+  mock_auths_.push_back(std::move(mock_auth));
+  mock_auth = std::make_unique<MockAuthenticator>();
+  EXPECT_CALL(*mock_auth.get(), doVerify(_, _, _))
+      .WillRepeatedly(
+          Invoke([](Http::HeaderMap&, std::vector<JwtLocationConstPtr>*,
+                    AuthenticatorCallback callback) { callback(Status::JwtUnknownIssuer); }));
+  mock_auths_.push_back(std::move(mock_auth));
+  createVerifier();
+
+  EXPECT_CALL(mock_cb_, onComplete(_)).WillOnce(Invoke([](const Status& status) {
+    ASSERT_EQ(status, Status::JwtUnknownIssuer);
```

Comment:
Why the second failure status is used for the final output instead of the first failure status?  can you add comment for that?